### PR TITLE
fix: Use the correct dockerhub image tag when building feature servers

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:
`sdk/python/tests/integration/online_store/test_universal_online.py::test_online_retrieval` has been failling consistently for a few PRs now. This PR should address the underlying issue.

**Which issue(s) this PR fixes**:

Fixes #

### CODE REVIEW

File: sdk/python/tests/integration/online_store/test_universal_online.py [hunk 1]
```diff
@@ -262,16 +265,17 @@ def _get_online_features_dict_remotely(
         request["features"] = features
     else:
         request["feature_service"] = features.name
-    for _ in range(25):
+    for attempt in range(25):
         # Send the request to the remote feature server and get the response in JSON format
         response = requests.post(
             f"{endpoint}/get-online-features", json=request, timeout=30
         ).json()
+        _logger.info(f"Attempt: {attempt}, {response}")
```
Comment (woop) [inline sdk/python/tests/integration/online_store/test_universal_online.py:273]: Should there be more detail here? Will our test logs just have a bunch of `attempt` statements without context?

Comment (achals) [reply sdk/python/tests/integration/online_store/test_universal_online.py:273]: This was for debugging in the github action; I'm going to remove this.

File: sdk/python/feast/infra/aws.py [hunk 2]
```diff
@@ -262,7 +264,9 @@ def _upload_docker_image(
         _logger.info(
             f"Pulling remote image {Style.BRIGHT + Fore.GREEN}{dockerhub_image}{Style.RESET_ALL}"
         )
-        for line in docker_client.api.pull(dockerhub_image, stream=True, decode=True):
+        for line in docker_client.api.pull(
+            dockerhub_image.replace("_", "."), stream=True, decode=True
+        ):
```
Comment (kevjumba) [inline sdk/python/feast/infra/aws.py:269]: Is this a new change(aws lambda docker image uses '_' instead of '.''s or did just never work?

**Review by kevjumba — APPROVED** [review]
/lgtm

File: sdk/python/feast/infra/aws.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/infra/aws.py]: This worked as of 0.16.0, but stopped working after that.

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**